invenio_swh/serializer.py

```diff
@@ -0,0 +1,152 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-SWH is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+"""Serializer for Invenio-SWH."""
+
+import xmltodict
+from flask_resources import MarshmallowSerializer
+from lxml import etree
+
+from invenio_swh.schema import SWHCodemetaSchema
+
+
+class BaseFormatter:
+    """Base formatter interface."""
+
+    def to_bytes(self, obj):
+        """Converts the object to bytes."""
+        raise NotImplementedError
+
+    def to_bytes_list(self, obj):
+        """Converts the object list to bytes."""
+        raise NotImplementedError
+
+    def to_str(self, obj):
+        """Converts the object to string."""
+        raise NotImplementedError
+
+    def to_str_list(self, obj):
+        """Converts the object list to string."""
+        raise NotImplementedError
+
+    def to_etree(self, obj):
```

Comment:
```suggestion
    def to_xml(self, obj):
```

minor: I know it's an `lxml.etree` under the hood, but just to keep things more human-friendly/readable